### PR TITLE
make life better

### PR DESCRIPTION
**Pull request by vendethiel**
(mostly for chrome)

wip, I still want to change 1/2 things

### CODE REVIEW

File: src/parser/Thread.co [hunk 1]
```diff
@@ -20,6 +20,9 @@ module.exports = class Thread
     # least
     for post of @posts
       for quoted of post.quotelinks
-        @post[quoted]backlinks.push post.no
+        if @post[quoted]
+          that.backlinks.push post.no
+        else
+          console.log "can't find post[#quoted]"
```
Comment (vendethiel) [inline src/parser/Thread.co]: not sure how that'd act

File: src/utils/dom.co [hunk 2]
```diff
@@ -11,7 +11,8 @@ export DOM = -> L \div |>
   &innerHTML = it
   &firstElementChild
 
-Element::matchesSelector ?= Element::mozMatchesSelector
+Element::matchesSelector ?= Element::mozMatchesSelector ?
```
Comment (vendethiel) [inline src/utils/dom.co:14]: I know you removed it but I'm afraid we need it ;)

**Comment by qqueue** [comment]
Thanks. The one other thing I need polyfilled for chrome is visibilityState. Do you think you could add that to this pull request?

File: src/parser/Thread.co (hunk 1 above)
Comment (qqueue) [inline src/parser/Thread.co]: Failing silently is fine. I broke backlinks in the last few commits unfortunately, sorry.

**Comment by vendethiel** [comment]
'course. I want to edit some other things, maybe they won't be chrome-relate. Browsed /a/ for a while until I got on a thread insulting SAO :p. (Valrave is insane crap imho, guunddddd)

**Comment by vendethiel** [comment]
Oh yeah watabout a `dist` dir containing the compiled js ? 4u2have moar users

File: src/parser/Thread.co (hunk 1 above)
Comment (vendethiel) [inline src/parser/Thread.co]: Erm, id you think a silent fail won't bring you more headaches than a console,log, sure :).

**Comment by qqueue** [comment]
> I got on a thread insulting SAO

>SAO
>good
pick 1.

> `dist` dir

See #9 for a better suggestion, IMHO.

**Comment by vendethiel** [comment]
> pick 1

Idc about the anime, it went full retard ~ep10. Loved the novel **so much** (tbh I think the best anime I've seen up to now is Steins;gate)

**Comment by vendethiel** [comment]
uuh chrome is the only one to have prefix on visibilitystate now ? eww

**Comment by qqueue** [comment]
> visibilityState on chrome

yup, shit sux. Good thing there won't be any more prefixes going forward.

---

I don't know if you can see it or not, but I broke your merge. Rebase on master and force push, if your OS can handle it.

**Comment by vendethiel** [comment]
> Get lost, chrome users

---

>  if your OS can handle it.

c'mon, you can do better than that

File: src/parser/Thread.co [hunk 3]
```diff
@@ -20,9 +20,7 @@ module.exports = class Thread
     # least
     for post of @posts
       for quoted of post.quotelinks
-        # quotelinks can contain cross-thread/board links, so guard against those.
-        # TODO is it useful that quotelinks contains cross links while
-        # backlinks doesn't?
-        @post[quoted]?backlinks.push post.no
+        if @post[quoted]
+          that.backlinks.push post.no
```
Comment (qqueue) [inline src/parser/Thread.co:27]: Use my verson with the comment

File: src/utils/dom.co (hunk 2 above)
Comment (qqueue) [inline src/utils/dom.co:14]: `matchesSelector` is now just `matches`:

http://dev.w3.org/2006/webapi/selectors-api2/#matches

Can you change it to polyfill to `match`?

**Comment by vendethiel** [comment]
Nami-Doc/wowboardhelpers@b076f26554b868956730b2c63f7631bfd36955a5
MUH PERFORMANCES (use that to cache baconjs, and everything will be a-okay)

**Pull request closed without merging**